### DISCUSSION TITLE
Contribute ?

### DISCUSSION THREAD

**Discussion by PhilippeZeitoun** [Jimskapt/rust-book-fr#249] [Q&A]
Hello,
I'm learning Rust and translating some chapters in French for my own usage. I was wondering how to contribute to the french translation ?
I don't understand how submit or translate the Rust Book.
Is there a FAQ, or a detailed process somewhere ?
Thanks

**Comment by Jimskapt** [comment]
Hello @wisevolk !

Sorry, I've missed your discussion and I've seen it only today 😑

# Translation

The whole book has been translated in french.

# Proofreading

We need to help for proofreading pages which name start with an `🚧` :

![image](https://user-images.githubusercontent.com/3988245/112748158-50f73780-8fba-11eb-8ab5-08f4bc46a26c.png)

You can have more informations how to proofreading with Git on [CONTRIBUTING.md](https://github.com/Jimskapt/rust-book-fr/blob/french-release/FRENCH/CONTRIBUTING.md).

But a simplier method is only using GitHub suggestions in Pull Requests like that :

> You can find link to Pull Request on top of each page :
> 
> ![image](https://user-images.githubusercontent.com/3988245/112748214-a29fc200-8fba-11eb-8710-0370cc407b52.png)
> 
> Then on this PR you can do your work in a new `review` :
> 
> ![image](https://user-images.githubusercontent.com/3988245/112748268-fc07f100-8fba-11eb-8c61-9625fe9131c3.png)

By the way, you can also find the Pull Requests with needed proofreading with the label :
[https://github.com/Jimskapt/rust-book-fr/pulls?q=is%3Aopen+is%3Apr+label%3A%22need+proofreading%22](https://github.com/Jimskapt/rust-book-fr/pulls?q=is%3Aopen+is%3Apr+label%3A%22need+proofreading%22)

# Updating from English book

The English book keep living and we need to bring down its updates in our French book.

It is a realtivly complex process with Git and it has been made even heavier because of the split of example code in `/listings/` files.

Please don't be afraid to tell me if you are missing informations 😉
